ams/shared.py

```
@@ -131,6 +133,32 @@ def wrapper(*args, **kwargs):
+                file=sys.stdout, disable=no_tqdm)
+    pbar.update(0)
+    return pbar
+
+
+def _update_pbar(pbar, current, total):
+    """Updates and closes the progress bar."""
+    perc = np.round(min((current / total) * 100, 100), 2)
+    if pbar.total is not None:  # Check if pbar is still valid
+        last_pc = pbar.n / pbar.total * 100  # Get current percentage based on updated value
+    else:
+        last_pc = 0
```

Ruff (0.11.9)

Use ternary operator `last_pc = pbar.n / pbar.total * 100 if pbar.total is not None else 0` instead of `if`-`else`-block

(SIM108)